### PR TITLE
Tm 4226 add maintain entry level page

### PR DESCRIPTION
**Pull request by fimbriano**
Dual Merge: 
- [API PR](https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/1081)
- [Mock PR](https://github.com/MetaPhase-Consulting/mock-fsbid/pull/509)

[Ticket](https://metaphase.atlassian.net/browse/TM-4226)

Added an entry level menu item.
Added maintain entry level position page 
- includes a table that shows data
- filters to be able to filter the data (only works on dev1)
- pagination to page through the data (only works on dev1)
**Note: to check a checkbox, you must click a checkbox then press spacebar to check/uncheck, then press enter to save.
**Note2: MC End Date is only editable if MC checkbox is checked.

### CODE REVIEW

File: src/Components/EntryLevelPage/MaintainELPositionsTable/MaintainELPositionsTable.jsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  const defaultColDef = {
+    sortable: false,
+  };
+
+  const columnTypes = {
+    customDate: {
+      extendsDataType: 'date',
+      baseDataType: 'date',
+      valueFormatter: params =>
+        // convert to `dd/mm/yyyy`
```
Comment (fimbriano) [inline src/Components/EntryLevelPage/MaintainELPositionsTable/MaintainELPositionsTable.jsx:72]:
```suggestion
        // convert to `mm/dd/yyyy`
```

File: package.json [hunk 2]
```diff
@@ -14,6 +14,8 @@
     "@react-pdf-viewer/default-layout": "^2.11.0",
     "@sweetalert/with-react": "^0.1.1",
     "@wojtekmaj/react-daterange-picker": "^3.4.0",
+    "ag-grid-community": "~31.1.0",
+    "ag-grid-react": "~31.1.0",
```
Comment (M00staff) [inline package.json:18]: 👍

File: src/Components/AdministratorPage/ManageEntryLevel/ManageEntryLevel.jsx [hunk 3]
```diff
@@ -16,7 +16,7 @@ const ManageEntryLevel = () => {
 
   const userSelections = useSelector(state => state.entryLevelSelections);
 
-  const elPositionsList = useSelector(state => state.entryLevelPositions);
+  const elPositionsList = useSelector(state => state.entryLevelPositions.results);
```
Comment (M00staff) [inline src/Components/AdministratorPage/ManageEntryLevel/ManageEntryLevel.jsx:19]: 👍

File: src/Components/EntryLevelPage/EntryLevelPage.test.jsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+      <EntryLevelPage />
+    </MemoryRouter></Provider>);
+    expect(wrapper).toBeDefined();
+  });
+
+  it('is defined', () => {
+    const wrapper = shallow(<EntryLevelPage />);
+    expect(wrapper).toBeDefined();
+  });
+});
```
Comment (M00staff) [inline src/Components/EntryLevelPage/EntryLevelPage.test.jsx:24]: wow - i dont think anyone has written a unit test here in a long time

File: src/Components/EntryLevelPage/MaintainELPositionsTable/MaintainELPositionsTable.jsx [hunk 5]
```diff
@@ -0,0 +1,121 @@
+import { useEffect, useRef, useState } from 'react';
+import { AgGridReact } from 'ag-grid-react';
```
Comment (M00staff) [inline src/Components/EntryLevelPage/MaintainELPositionsTable/MaintainELPositionsTable.jsx:2]: nifty! 👍

File: tsconfig.json [hunk 6]
```diff
@@ -14,8 +14,8 @@
     // "declarationMap": true,                      /* Generates a sourcemap for each corresponding '.d.ts' file. */
     "sourceMap": true,                              /* Generates corresponding '.map' file. */
     // "outFile": "./",                             /* Concatenate and emit output to single file. */
-    // "outDir": "./",                              /* Redirect output structure to the directory. */
-    // "rootDir": "./",                             /* Specify the root directory of input files. Use to control the output directory structure with --outDir. */
+    // "outDir": "./dist",                              /* Redirect output structure to the directory. */
+    // "rootDir": "./src",                             /* Specify the root directory of input files. Use to control the output directory structure with --outDir. */
```
Comment (M00staff) [inline tsconfig.json:18]: what are all the updates to this file doing?

**Review by M00staff — APPROVED** [review]
this is great stuff!

all the patterns match up to the way we've been doing things, so kudos for picking it all up so quickly!

👍

**Pull request merged**